### PR TITLE
AAKD, I want linting in CI on the auto ML project

### PR DESCRIPTION
**Pull request by crsegerie**
Added pre-commit and ad ci pipeline verifying the same pre-commit.

The pre-commit includes black and flake8. Lines have been fixed to max 100 characters.

The ci/cd pipeline in the GitHub action uses the same file as the pre-commit in local.

### CODE REVIEW

**Comment by crsegerie** [comment]
end 2 end tests passing

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+## Help us to help you!
+
```
Comment (PierreLeveau) [inline contributing.md:2]: install grammarly to fix typos and syntax errors.

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

**Comment by PierreLeveau** [comment]
also trying locally to see if the installation procedure works.

**Comment by crsegerie** [comment]
I've tested the black formatting, and even with a correctable modification, black fails in the GitHub action. So it's working as expected.

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+## Help us to help you!
```
Comment (PierreLeveau) [inline contributing.md:1]: For now, you can keep the contributing.md to the minimum (no policy yet about external contributions).

```
# Contributing 

## Development environment
To install the development environment, please follow these steps:
```bash
pip install -r requirements_dev.txt
pre-commit install
pre-commit run --all-files
```

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

**Review by PierreLeveau — APPROVED** [review]

**Pull request merged**